enhancements/network/on-prem-dual-stack-vips.md

```diff
@@ -0,0 +1,421 @@
+changes needed in that configuration. We'll need to verify that as part of
+this implementation though.
+
+We could also modify the apiserver configuration so it listens on both v4 and
+v6, but since there's currently no way to set up the `kubernetes.default`
+Service as dual stack this doesn't accomplish much. Traffic to the
+secondary IP version will have to come in via the VIP, which already has
+haproxy to handle both versions.
+
+### baremetal-runtimecfg
+
+There are a number of places in runtimecfg where a VIP is used to determine
+which ip version to use (for example, when deciding whether to use the ipv4
+or ipv6 addresses in the unicast_peer list). This logic will need to be
+updated to handle the fact that there may be both v4 and v6 VIPs.
+
+Additionally, the new VIPs will need to be wired in to the rendering code.
+
+### Open Questions [optional]
+
+Asked and answered.
+
+### Test Plan
+
+Most of this feature will be covered by the existing dual stack test jobs.
+A few added tests will be needed to verify the functionality of the new VIPS.
+
+### Graduation Criteria
+
+We do not anticipate needing a graduation process for this feature. The
+internal loadbalancer implementation has been around for a number of
+releases at this point and we are just extending it.
+
+#### Dev Preview -> Tech Preview
+
+NA
+
+#### Tech Preview -> GA
+
+NA
+
+#### Removing a deprecated feature
+
+NA
+
+### Upgrade / Downgrade Strategy
+
+Upgrades and downgrades will be handled the same way they are for the current
+internal loadbalancer implementation. On upgrade, existing VIP configurations
+will be maintained. We will not (and cannot) automatically add new VIPs to a
+cluster on upgrade. If a deployer of an older dual stack cluster wants the new
+VIP functionality that will have to be a separate operation from upgrade.
+
+### Version Skew Strategy
+
+This will also be handled as it is today. The keepalived configuration used
+for the new version must work with the old version until all nodes have been
+upgraded. If breaking changes are needed, there will be a migration mechanism
+that runs after upgrade is complete (similar to what we did in the multicast
+to unicast migration).
+
+### Operational Aspects of API Extensions
+
+NA
+
+#### Failure Modes
+
+NA
+
+#### Support Procedures
+
+NA
+
+## Implementation History
+
+4.11: Initial implementation
+
+## Drawbacks
+
+Nothing significant. A very small amount of compute resources will be used
+to manage the new VIPs. Even this can be avoided by simply not specifying
+the second VIP if it is not needed.
+
+## Alternatives
+
+Implement this somewhat like the
+[existing workaround](https://github.com/yboaron/kepalived-v6) - create a
+separate instance of keepalived that is responsible for ipv6 only. This
+results in some duplication of containers and configs, but it would probably
+simplify the code changes.
+
+### Alternative install-config layouts
+
+#### Just add v6 versions of the VIP fields
```

Comment:
This alternative seems like it is the most unambiguous to the user. 

While describing the preferred solution, you mentioned that it would be beneficial to mark one address as the primary (and the other as secondary). So, in that case, for this alternative, would we also need another optional config item to point to the promaryVIP? For example, primaryApiVIP could have the values apiVIP or apiVIPv6. If unspecified it defaults to apiVIP.